.github/workflows/puppet-test.yml

```diff
@@ -160,25 +280,59 @@ jobs:
       - name: Install pytest
         run: pip install -r "${{ env.REQUIREMENTS_PATH }}"
 
+      - name: Start local MSI server
+        shell: powershell
+        run: |
+          # Start Python HTTP server in background to serve MSI files from dist directory
+          Start-Process python -ArgumentList "-m http.server 8000 --directory dist" -WindowStyle Hidden
+          
+          # Wait for server to be ready with retry logic
+          $maxAttempts = 10
+          $attempt = 1
+          $serverReady = $false
+          
+          while ($attempt -le $maxAttempts) {
+            Write-Host "Checking if server is ready (attempt $attempt/$maxAttempts)..."
+            try {
+              $response = Invoke-WebRequest -Uri "http://localhost:8000/" -UseBasicParsing -TimeoutSec 2
+              if ($response.StatusCode -eq 200) {
+                Write-Host "Local MSI server started successfully"
+                $serverReady = $true
+                break
+              }
+            } catch {
+              Write-Host "Server not ready yet, waiting..."
+            }
+            Start-Sleep -Seconds 1
+            $attempt++
+          }
+          
+          if (-not $serverReady) {
+            Write-Error "Failed to start local MSI server after $maxAttempts attempts"
+            exit 1
+          }
+
       - name: Test puppet deployment
         id: pytest
         continue-on-error: true
+        shell: powershell
```

Comment:
Just curious Q.: was this just to make it explicit or if there is something that needs `powershell` instead of the default `pwsh` shell?